src/plugins/data/common/search/expressions/esaggs/request_handler.ts

```diff
@@ -113,52 +127,19 @@ export const handleRequest = async ({
   requestSearchSource.setField('filter', filters);
   requestSearchSource.setField('query', query);
 
-  let request;
-  if (inspectorAdapters.requests) {
-    inspectorAdapters.requests.reset();
-    request = inspectorAdapters.requests.start(
-      i18n.translate('data.functions.esaggs.inspector.dataRequest.title', {
-        defaultMessage: 'Data',
-      }),
-      {
-        description: i18n.translate('data.functions.esaggs.inspector.dataRequest.description', {
-          defaultMessage:
-            'This request queries Elasticsearch to fetch the data for the visualization.',
-        }),
-        searchSessionId,
-      }
-    );
-    request.stats(getRequestInspectorStats(requestSearchSource));
-  }
-
-  try {
-    const response = await requestSearchSource.fetch({
-      abortSignal,
-      sessionId: searchSessionId,
-    });
-
-    if (request) {
-      request.stats(getResponseInspectorStats(response, searchSource)).ok({ json: response });
-    }
+  inspectorAdapters.requests?.reset();
+  const requestResponder = getRequestMainResponder(inspectorAdapters, searchSessionId);
 
-    (searchSource as any).rawResponse = response;
-  } catch (e) {
-    // Log any error during request to the inspector
-    if (request) {
-      request.error({ json: e });
-    }
-    throw e;
-  } finally {
-    // Add the request body no matter if things went fine or not
-    if (request) {
-      request.json(await requestSearchSource.getSearchRequestBody());
-    }
-  }
+  const response$ = await requestSearchSource.fetch$({
+    abortSignal,
+    sessionId: searchSessionId,
+    requestResponder,
```

Comment:
i think it makes sense here to pass `inspectorAdapters.requests`  in, as we'll need that to construct the other bucket inside searchsource as well. i'm fine with leaving it as is for now however as we plan to move the other bucket code inside searchsource in a followup pr.

Reply:
I agree and I intend to make that change in the other bucket PR